aiojobs/_scheduler.py

```diff
@@ -71,8 +81,19 @@ def closed(self):
         if self._closed:
             raise RuntimeError("Scheduling a new job after closing")
         job = Job(coro, self, self._loop)
-        should_start = (self._limit is None or
-                        self.active_count < self._limit)
+        should_start = (self._limit is None
+                        or self.active_count < self._limit)
+        should_wait = (self._pending_limit is not None
+                       and self.pending_count >= self._pending_limit)
+        if not should_start and should_wait:
+            waiter = self._loop.create_future()
+            waiter._job = job
```

Comment:
No chance to accept this line: adding a custom attribute to asyncio Future has very bad smell.